app/helpers/validation_request_helper.rb

```diff
@@ -58,6 +58,14 @@ def cancel_request_url(planning_application, validation_request)
          validation_request)
   end
 
+  def edit_validation_request_url(validation_request)
+    link_to "Edit",
+            send("edit_planning_application_#{request_type(validation_request)}_path", validation_request.planning_application,
+                 validation_request)
+  rescue NoMethodError
```

Comment:
Using exceptions like this isn't a good pattern - better to check the value returned by `request_type` is valid, e.g.

``` ruby
def edit_validation_request_url(validation_request)
  type = request_type(validation_request)
  planning_application = validation_request.planning_application

  unless type == :replacement_document
    link_to "Edit", [:edit, planning_application, validation_request]
  end
end
```

The array in `link_to` triggers polymorphic routing:
https://api.rubyonrails.org/classes/ActionDispatch/Routing/PolymorphicRoutes.html

Reply:
thanks, change made